hrjobcontract/js/src/job-contract/controllers/form/form-leave.js

```diff
@@ -0,0 +1,38 @@
+define([
+    'job-contract/vendor/fraction',
+    'job-contract/controllers/controllers'
+], function (Fraction, controllers) {
+    'use strict';
+
+    controllers.controller('FormLeaveCtrl',['$scope', '$rootScope', '$filter', '$log', '$q', 'UtilsService',
+      function ($scope, $rootScope, $filter, $log, $q, UtilsService) {
+        $log.debug('Controller: FormLeaveCtrl');
+
+        var promiseNumberOfPublicHolidays = UtilsService.getNumberOfPublicHolidaysInCurrentPeriod();
+
+        promiseNumberOfPublicHolidays.then(function(result){
+          var number = parseFloat(result);
+          if(isNaN(number)) {
+            number = 0;
+          }
+          $scope.numberOfPublicHolidays = number;
+        });
+
+        $scope.$watch('entity.leave', function(newValue, oldValue) {
+          var newTypeWithPublicHolidays = null;
+          newValue.forEach(function(value, i) {
+            if(value.add_public_holidays === '1' && oldValue[i].add_public_holidays === '0') {
+              newTypeWithPublicHolidays = value.leave_type;
+              return false;
+            }
+          });
+          if(newTypeWithPublicHolidays) {
+            newValue.forEach(function(value) {
+              if(value.leave_type != newTypeWithPublicHolidays) {
+                value.add_public_holidays = '0';
+              }
+            });
+          }
+        }, true);
```

Comment:
entity.leave is an array of objects and we need to listen to when the properties of these objects change. $watchCollection wasn't able to do this.